modules/home-manager/alacritty.nix

```diff
@@ -6,7 +6,7 @@
 let
   inherit (lib) ctp;
   cfg = config.programs.alacritty.catppuccin;
-  enable = cfg.enable && config.programs.alacritty.enable;
+  enable = (cfg.enable && config.programs.alacritty.enable) || config.catppuccin.enable) || config.catppuccin.enable;
```

Comment:
I would like to have some guidance on this.

Reply:
Its a bit rushed buy maybe something like this
`isCatppuccinEnabled = conf: prog: conf.catppuccin.enable && conf.programs.${prog}.enable && conf.programs.${prog}.catppuccin.enable;`

Reply:
I'll try something.

Reply:
I think something like this would work:
```nix
# lib/default.nix
{config, ...}: {
	# ...
	isEnabled = cfg: config.catppuccin.enable || cfg.enable && cfg.catppuccin.enable; 
}

# modules/home-manager/alacritty.nix
{config, lib, ...}:
let
	inherit (lib) ctp;
	enable = ctp.isEnabled config.programs.alacritty;
in { ... }
```

But then, you're leaving out the option for users to disable theming as an override. IMO the more reasonable thing to do is to just make the default for each enable option to be `config.catppuccin.enable`, [which I believe is already the case](https://github.com/catppuccin/nix/blob/75663896d0c16cd59d567f21f091b1c9338d7118/modules/lib/default.nix#L16)

Reply:
good catch, totally forgot to add the all enabled case.

Reply:
I just pushed something, let me know what you think.

Reply:
Very cool idea, I just pushed everything again ;)